### PR TITLE
The recipe must be initialised in the critical path. r=azasypkin

### PR DESCRIPTION
**Pull request by gmarty**
If not, landing on a the Recipes page when refreshing the app won't work.

### CODE REVIEW

File: app/js/lib/foxbox/foxbox.js [hunk 1]
```diff
@@ -89,15 +89,15 @@ export default class Foxbox extends Service {
...
-        settings: this[p.settings],
-        net: this[p.net]
-      });
     });
 
     return this._initUserSession()
       .then(() => {
+        this.recipes = new Recipes({
+          settings: this[p.settings],
+          net: this[p.net]
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: Looks good, the only thing that worries me a bit is that ```net``` is initialized once discovery is resolved.

**Comment by azasypkin** [comment]
r+, just one concern, but maybe it's okay for now.

**Comment by gmarty** [comment]
I reorganised the sequence to initialise the network first then the rest as of today.

**Comment by azasypkin** [comment]
Hmm, looks like I'm facing similar thing in https://github.com/fxbox/app/issues/116.

This "chicken and the egg problem" blocking me right now, but I have one idea in mind that I'm trying to implement.

**Comment by azasypkin** [comment]
> but I have one idea in mind that I'm trying to implement.

Okay it works well with DB, but not with Network :/ For DB idea is to keep internal initialization promise that is used by all async operations, so you can call any async function on DB and it will be performed and resolved only when initialization promise is resolved - so it safe to call such methods at any time. But Network is different. Wondering if should just make Network simple and make Foxbox to manage local/tunnel origin...

**Comment by gmarty** [comment]
We can close this one now that https://github.com/fxbox/app/pull/120 landed.

**Pull request closed without merging**